generator/src/main/java/com/scottlogic/deg/generator/walker/reductive/FieldCollection.java

```diff
@@ -123,9 +124,16 @@ private FixedField getFixedFieldWithValuesForField(Field field, ConstraintNode r
             .stream()
             .filter(c -> c.getField().equals(field))
             .collect(Collectors.toSet());
+        Set<AtomicConstraint> constraintsForDecisions = rootNode.getDecisions()
+            .stream()
+            .flatMap(d -> d.getOptions().stream())
```

Comment:
This process is not fully recursive, can it use the changes from @pdaulbyscottlogic now?

Reply:
As discussed with Paul the implementation required for this change wasn't easily supported by the changes that were introduced.

Reply:
Changed to a fully-recursive algorithm, employing the recursive visitor